Client/Application/Shortcuts/ActivityShortcutManager.swift

```diff
@@ -0,0 +1,203 @@
+                        settingsNavigationController.modalPresentationStyle =
+                            UIDevice.current.userInterfaceIdiom == .phone ? .pageSheet : .formSheet
+                        settingsNavigationController.navigationBar.topItem?.leftBarButtonItem =
+                            UIBarButtonItem(barButtonSystemItem: .done, target: settingsNavigationController, action: #selector(settingsNavigationController.done))
+                        
+                        // All menu views should be opened in portrait on iPhones.
+                        UIDevice.current.forcePortraitIfIphone(for: UIApplication.shared)
+
+                        bvc.present(settingsNavigationController, animated: true)
+                    case .installed(let connected):
+                        if !connected {
+                            BraveVPN.reconnect()
+                        }
+                }
+            case .openBraveToday:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+
+                guard let newTabPageController = bvc.tabManager.selectedTab?.newTabPageViewController else { return }
+                newTabPageController.scrollToBraveToday()
+            case .openPlayList:
+                let playlistController = (UIApplication.shared.delegate as? AppDelegate)?.playlistRestorationController ?? PlaylistViewController()
+                playlistController.modalPresentationStyle = .fullScreen
+                
+                bvc.present(playlistController, animated: true)
+        }
+    }
+    
+    // MARK: Intent Creation Methods
+    
+    private func createCustomIntent(for type: IntentType, with urlString: String) -> INIntent {
+        switch type {
+            case .openWebsite:
+                let intent = OpenWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenWebsiteSuggestedPhrase
+                
+                return intent
+            case .openHistory:
+                let intent = OpenHistoryWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenHistorySuggestedPhrase
+                
+                return intent
+            case .openBookmarks:
+                let intent = OpenBookmarkWebsiteIntent()
+                intent.websiteURL = urlString
+                intent.suggestedInvocationPhrase = Strings.Shortcuts.customIntentOpenBookmarkSuggestedPhrase
+                
+                return intent
+        }
+    }
+    
+    // MARK: Intent Donation Methods
+    
+    public func donateCustomIntent(for type: IntentType, with urlString: String) {
+        let intent = createCustomIntent(for: type, with: urlString)
+
+        let interaction = INInteraction(intent: intent, response: nil)
+        interaction.donate { (error) in
```

Comment:
brackets around `error` can be removed

Reply:
Fix at https://github.com/brave/brave-ios/pull/3579/commits/e1e25adef24913c2fa02c7416101415ae1b23053